c2corg_api/models/feed.py

```diff
@@ -142,3 +142,82 @@ class DocumentChange(Base):
             postgresql_using='gin'),
         Base.__table_args__
     )
+
+# For performance reasons, areas and users are referenced in simple integer
+# arrays in 'feed_document_changes', no PK-FK relations are set up. To prevent
+# inconsistencies, triggers are used.
+trigger_ddl = DDL("""
+-- when creating a change, check that the given user and area ids are valid
+CREATE OR REPLACE FUNCTION guidebook.check_feed_ids() RETURNS TRIGGER AS
+$BODY$
+DECLARE
+  user_id int;
+  area_id int;
+BEGIN
+  -- check user ids
+  FOREACH user_id IN ARRAY new.user_ids LOOP
+    PERFORM id from users.user where id = user_id;
+    IF NOT FOUND THEN
+      RAISE EXCEPTION 'Invalid user id: %%', user_id;
+    END IF;
+  END LOOP;
+  -- check area ids
+  FOREACH area_id IN ARRAY new.area_ids LOOP
+    PERFORM document_id from guidebook.areas where document_id = area_id;
+    IF NOT FOUND THEN
+      RAISE EXCEPTION 'Invalid area id: %%', area_id;
+    END IF;
+  END LOOP;
+  RETURN null;
+END;
+$BODY$
+language plpgsql;
+
+CREATE TRIGGER guidebook_feed_document_changes_insert
+AFTER INSERT OR UPDATE ON guidebook.feed_document_changes
+FOR EACH ROW
+EXECUTE PROCEDURE guidebook.check_feed_ids();
+
+-- when deleting a user, check that there are no changes referencing the user
```

Comment:
This trigger is run when a row from the user table is deleted. In case of an exception, the transaction is canceled (the user row is not deleted).